perf/cachingoptimizer.jl

```diff
@@ -0,0 +1,24 @@
+# Current output:
+#  38.889 ns (4 allocations: 112 bytes)
```

Comment:
These numbers aren't particularly helpful given that nobody else has access to the machine where this was timed. I think the link to the issue is sufficient.